### PR TITLE
Matrix(double[][] A) constructor - #1

### LINKED ISSUE #1: Matrix(double[][] A) constructor
Construct a matrix from a 2-D array.

### PR DESCRIPTION
**Pull request by KaraBeason**
Closes #1.  Added constructor that copies 2D array into matrix.  Unit tests pass, style check passes.  Admittedly I forgot to write the test first, but I will remember to do that from here on out!

### CODE REVIEW

**Review by haietza — CHANGES_REQUESTED** [review]
This constructor needs to throw an IllegalArgumentException. Add test for exception being thrown as well. I would also use m and n variables in test for consistency. Everything else looks good!

**Comment by KaraBeason** [comment]
@haietza can you give me more information on the case in which it would throw this exception?

Edit: NVM, saw the API.  Fixing now!

**Comment by KaraBeason** [comment]
@haietza for the test, can you specify which line you're talking about?  Line 35 or the for loop?

**Comment by haietza** [comment]
I initialized them at the start of my test and then used them in the array
initialization and in the for loop. Not major, just a suggestion to keep
our code consistent?

Michelle Melton
Web Developer/Programmer
Center for Academic Excellence

**Comment by KaraBeason** [comment]
@haietza Sorry, realized what you meant as soon as I commented about the variables (Seems to be a trend with me today - stupid brain!).  Yes, that makes sense to me!  Thank you!  Everything should be resolved now, I think.  LMK what you think!

File: Javatrix.java [hunk 1]
```diff
@@ -25,6 +25,31 @@
     int n;
 
     /**
+     * Construct a matrix from a 2D array.
+     *
+     * @param arr - 2D array to be copied into matrix
+     **/
```
Comment (haietza) [inline Javatrix.java]: Doesn't this need an @throws tag?

File: Javatrix.java [hunk 2]
```diff
@@ -25,6 +25,31 @@
...
+    public Javatrix(double[][] arr)
```
Comment (haietza) [inline Javatrix.java]: Doesn't this need a "throws IllegalArgumentException"?

File: Javatrix.java [hunk 3]
```diff
@@ -25,6 +25,31 @@
...
+        this.m = arr.length;
+        this.n = arr[0].length;
+        matrix = new double[m][n];
+        for (int i = 0; i < m; i++)
+        {
+            if (arr[i].length != n)
+            {
+                String ex = "All rows must be the same length.";
+                throw new IllegalArgumentException(ex);
+
```
Comment (haietza) [inline Javatrix.java]: Delete extra line?

File: JavatrixTest.java [hunk 4]
```diff
@@ -24,6 +24,46 @@
...
+    {
+        m = 2;
+        n = 2;
+        matrix = new double[m][];
+        matrix[0] = new double[n];
+        matrix[1] = new double[3];
+
+        Javatrix actual = new Javatrix(matrix);
+    }
+
```
Comment (haietza) [inline JavatrixTest.java]: Delete extra line?

File: JavatrixTest.java [hunk 5]
```diff
@@ -24,6 +24,46 @@
     private double[][] matrix;
     private int m;
     private int n;
+
+
```
Comment (haietza) [inline JavatrixTest.java]: Delete extra line?

File: JavatrixTest.java [hunk 6]
```diff
@@ -24,6 +24,46 @@
...
+        {
+            for (int j = 0; j < n; j++)
+            {
+                matrix[i][j] = java.lang.Math.random() * 10 + 1; 
+            }
+        }
+        
+        Javatrix actual = new Javatrix(matrix);
+        assertArrayEquals(matrix, actual.matrix);
+
```
Comment (haietza) [inline JavatrixTest.java]: Delete extra line?

**Comment by haietza** [comment]
No worries - I have been doing that just about every time I commit and push
- I think of like 6 things I forgot! TGIF :).

Michelle Melton
Web Developer/Programmer
Center for Academic Excellence

**Review by haietza — CHANGES_REQUESTED** [review]
Let me know what you think about these things...thanks!

**Comment by haietza** [comment]
@caldwellc1 @KaraBeason Question - I had not been including the .class files in my commits because I think I remember Dr Fenwick not doing that in our previous assignment. Do either of you know what standard protocol is on this?

**Comment by haietza** [comment]
@caldwellc1 @KaraBeason I have been using the method headers from the API, matching their naming convention for parameters, etc. Should we try to make all our code consistent in this way?

**Comment by KaraBeason** [comment]
@haietza I think that's a good idea.  I don't see method headers in the API .... could you link me please?

ETA: I removed the class files from my commit, I think you are right that it is best practice.  I think I got everything this time!  LMK!

**Comment by haietza** [comment]
I was just using the headers in the API Dr Fenwick provided: http://cs.appstate.edu/~jbf/classes/cs5666/Javatrix/. But I just discovered that some of those won't pass checkstyle (using A for a parameter name), so I think as long as we are consistent between the team, we should be good :).

**Review by haietza — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**